### PR TITLE
virt-handler: mount all of /var/lib/kubelet

### LINKED ISSUE #12520: cpumanager not detected when enabled
**What happened**:
I enabled CPUManager feature gate and enabled it in the kubelet of one node, and verified that /proc/1/root/var/lib/kubelet/cpu_manager_state contains {"policyName":"static",... } after restarting the kubelet. However, the virt-handler keeps reporting "failed to set a cpu manager label on host host","pos":"heartbeat.go:174","reason":"no cpumanager policy file found"

**What you expected to happen**:
The cpumanager state file that exists is found and the cpumanager label is set to true

**How to reproduce it (as minimally and precisely as possible)**:
Enable CPUManager feature gate, enable CPU manager in kubelet of one node.

**Additional context**:
I tried rebooting the node, restarting kubelet, restarting virt-handler, didn't seem to change anything.

**Environment**:
- KubeVirt version (use `virtctl version`): v1.2.0-beta.0
- Kubernetes version (use `kubectl version`): v1.23.7
- VM or VMI specifications: N/A
- Cloud provider or hardware configuration: N/A
- OS (e.g. from /etc/os-release): Ubuntu 22.04
- Kernel (e.g. `uname -a`): 5.15.0-117-generic
- Install tools: N/A
- Others: N/A

### PR DESCRIPTION
**Pull request by jean-edouard**
### What this PR does
Before this PR:
Only `/var/lib/kubelet/pods` is mounted in virt-handler, the rest of /var/lib/kubelet is accessed through `/proc/1/root`

After this PR:
`/var/lib/kubelet` is mounted in virt-handler, which lifts the need for /proc/1/root and should correctly follow an eventual symlink, see issue.

Fixes #12520 

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
/var/lib/kubelet on the nodes can now be a symlink
```

### CODE REVIEW

**Comment by alicefr** [comment]
I wonder why we were using `/proc/1/root` in the first place, do you know?

**Comment by jean-edouard** [comment]
> I wonder why we were using `/proc/1/root` in the first place, do you know?

It's basically the default way for virt-handler to access the node filesystem.
It usually works well, until you create symlinks there...

File: pkg/virt-operator/resource/generate/components/daemonsets.go [hunk 1]
```diff
@@ -265,19 +266,18 @@ func NewHandlerDaemonSet(namespace, repository, imagePrefix, version, launcherVe
 	attachProfileVolume(pod)
 
 	bidi := corev1.MountPropagationBidirectional
-	// NOTE: the 'kubelet-pods-shortened' volume mounts the same host path as 'kubelet-pods'
-	// This is because that path holds unix domain sockets. Domain sockets fail when they're over
-	// ~ 100 characters, so that shortened volume path is to allow domain socket connections.
+	// NOTE: the 'kubelet-pods' volume mount exists because that path holds unix socket files.
+	// Socket files fail when their path is longer than 108 characters,
+	//   so that shortened volume path is to allow domain socket connections.
```
Comment (iholder101) [inline pkg/virt-operator/resource/generate/components/daemonsets.go:271]: tiny nit: redundant spaces

**Review by iholder101 — COMMENTED** [review]
Thank you @jean-edouard!

Another tiny nit is this comment which is now misleading:
https://github.com/kubevirt/kubevirt/blob/0d1c5a274cd297b3742a82ddfdce0de8df0e22c0/pkg/virt-handler/seccomp/seccomp.go#L14-L15

File: pkg/virt-operator/resource/generate/components/daemonsets.go (hunk 1 above)
Comment (jean-edouard) [reply pkg/virt-operator/resource/generate/components/daemonsets.go:271]: It's a commenting style I like, those extra spaces show we're in the middle of a sentence.
I can delete them if that's against our coding style.

**Comment by jean-edouard** [comment]
> > > I wonder why we were using `/proc/1/root` in the first place, do you know?
> > 
> > 
> > It's basically the default way for virt-handler to access the node filesystem. It usually works well, until you create symlinks there...
> 
> Do we have any ideas why symlinks do not work in this case?

Yeah, all symlinks that point to an absolute path will resolve to the container rootfs instead of `/proc/1/root`.
Like, if the host symlinks `/var/lib/kubelet` to `/kubelet`, when virt-handler resolves `/proc/1/root/var/lib/kubelet` it will look for `/kubelet` and NOT `/proc/1/root/kubelet`.
In this case, doing a relative symlink of `/var/lib/kubelet` to `../../kubelet` should be a valid workaround, but with this PR both cases will work.

File: pkg/virt-operator/resource/generate/components/daemonsets.go (hunk 1 above)
Comment (iholder101) [reply pkg/virt-operator/resource/generate/components/daemonsets.go:271]: Sure, thanks for elaborating :+1:

**Comment by iholder101** [comment]
> > > > I wonder why we were using `/proc/1/root` in the first place, do you know?
> > > 
> > > 
> > > It's basically the default way for virt-handler to access the node filesystem. It usually works well, until you create symlinks there...
> > 
> > 
> > Do we have any ideas why symlinks do not work in this case?
> 
> Yeah, all symlinks that point to an absolute path will resolve to the container rootfs instead of `/proc/1/root`. Like, if the host symlinks `/var/lib/kubelet` to `/kubelet`, when virt-handler resolves `/proc/1/root/var/lib/kubelet` it will look for `/kubelet` and NOT `/proc/1/root/kubelet`. In this case, doing a relative symlink of `/var/lib/kubelet` to `../../kubelet` should be a valid workaround, but with this PR both cases will work.

Interesting! Makes sense :)
Thank you @jean-edouard!

/approve

**Pull request merged**